From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aq17 <aqiu@pulumi.com>
Date: Thu, 25 May 2023 10:33:38 -0700
Subject: [PATCH] Make shared features config optional

SharedFeatures config added in https://github.com/hashicorp/terraform-provider-azurerm/commit/f860827bcb64a4f6f97274e6ee276dd54b9bd4e2
Modified in fork in https://github.com/pulumi/terraform-provider-azurerm/commit/6ac9e0b5ff9d72272d76f311a7665c03b12cb722

This originally included a conditional block to only set the SharedFeatures config if it had a value but this seems to have been lost during the provider config being refactored in https://github.com/hashicorp/terraform-provider-azurerm/commit/57bcf134131514ffcc962f20b12f993fad7d6060#diff-58d6a027753b50994deb7e11e4a99dde423f35844986019bd9cea5e0c94aba22

diff --git a/internal/provider/features.go b/internal/provider/features.go
index 25b07caab5..9837e1c709 100644
--- a/internal/provider/features.go
+++ b/internal/provider/features.go
@@ -434,7 +434,7 @@ func schemaFeatures(supportLegacyTestSuite bool) *pluginsdk.Schema {
 
 	return &pluginsdk.Schema{
 		Type:     pluginsdk.TypeList,
-		Required: true,
+		Optional: true,
 		MaxItems: 1,
 		MinItems: 1,
 		Elem: &pluginsdk.Resource{
